test/modify-status.spec.js

```diff
@@ -0,0 +1,133 @@
+var chance = require('chance')();
+var chai = require('chai');
+var rewire = require('rewire');
+var sinon = require('sinon');
+var emoji = require('node-emoji');
+
+chai.use(require('chai-as-promised'));
+
+var modifyStatus = rewire('../src/modify-status');
+
+describe('Handling request', () => {
+	var fakeSlackApi, token, text, status;
+	var act = () => modifyStatus(text, token);
+	var refreshMocks = () => {
+		fakeSlackApi = {
+			getUserInfo: sinon.stub(),
+			sendResponse: sinon.stub()
+		};
+		modifyStatus.__set__({
+			slackApi: fakeSlackApi
+		});
+	}
+
+	describe('When processing a request', () => {
+		beforeEach(() => {
+			token = chance.string();
+			text = chance.sentence();
+			status = {
+				status_text: 'Working from home',
+				status_emoji: ':house_with_garden:'
+			}
+			refreshMocks();
+		});
+		describe('And the user passes in a nonempty text', () => {
+					var randomEmoji = emoji.random().emoji;
```

Comment:
Generating your `randomEmoji` should go in a `beforeEach()` otherwise you are reusing the same emoji over and over...this is a little gotcha from scoping..the reason this happens is because the way the test runner works it actually executes all the test files so any declarations or code happening outside of a `before()`, `act()` or `it()` will happen when the tests initialize which only happens once...from there the test runner can gather the "execution plan" to execute all the `before()` then execute the `act()` then execute the `it()...so the code would look like
```
var randomEmoji;
beforeEach(() => randomEmoji = emoji.random(););
```
- [ ] ensure a random emoji is generated for every test.